### PR TITLE
feat(rpc): improve reorg checks

### PR DESCRIPTION
**Pull request by davidtaikocha**

### CODE REVIEW

**Review by mask-pp — APPROVED** [review]

File: pkg/rpc/methods.go [hunk 1]
```diff
@@ -483,6 +502,119 @@ func (c *Client) CheckL1ReorgFromL2EE(ctx context.Context, blockID *big.Int) (bo
...
+
+	if currentRoot != l1SignalRoot {
+		log.Info("Reorg detected due to L1 signal root mismatch", "blockID", blockID)
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (c *Client) getSyncedL1SnippetFromAnchcor(
```
Comment (cyberhorsey) [inline pkg/rpc/methods.go:557]: typo here, anchor is mispelled

File: prover/prover.go [hunk 2]
```diff
@@ -64,6 +64,7 @@ type Prover struct {
 	l1Current              *types.Header
 	reorgDetectedFlag      bool
 	tiers                  []*rpc.TierProviderTierWithID
+	l1SingalService        common.Address
```
Comment (cyberhorsey) [inline prover/prover.go:67]: typo here, signal is mispelled

**Review by cyberhorsey — CHANGES_REQUESTED** [review]

File: pkg/rpc/methods.go (hunk 1 above)
Comment (davidtaikocha) [reply pkg/rpc/methods.go]: fixed

File: prover/prover.go (hunk 2 above)
Comment (davidtaikocha) [reply prover/prover.go]: fixed

**Pull request merged**